tlvf: add support for optional MultiAP-Ext in M1

tomereli opened the pull request:
This is a first attempt at adding support for an optional
MultiAP-Extension attribute in the vendor extension of M1/M2.

First, sWscAttrVersion2 is renamed to sWscVExtAttrVersion2. It will
not be used as part of the M1 anymore, but it may still be used
somewhere else, like in the M2.

Second, a new sWscAttrVersion2 attribute is added. It contains only
the Version2 subitems, and not the whole vendor extension.

A new class is also added: cWscVendorExt. It contains both the vendor
extension fields, and unknown length list vs_data, which may contain
the version2 and/or MultiAP-Extension items.

son_slave_thread.cpp also needs to be modified. Now that the vendor
extension uses an unknown length list, it no longer contains the
default value for the Version2 attribute, so they have to be added
manually.

Fixes #490
Fixes #213

Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>
Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

rmelotte committed:
tlvf: add support for optional MultiAP-Ext in M1

This is a first attempt at adding support for an optional
MultiAP-Extension attribute in the vendor extension of M1/M2.

First, sWscAttrVersion2 is renamed to sWscVExtAttrVersion2. It will
not be used as part of the M1 anymore, but it may still be used
somewhere else, like in the M2.

Second, a new sWscAttrVersion2 attribute is added. It contains only
the Version2 subitems, and not the whole vendor extension.

A new class is also added: cWscVendorExt. It contains both the vendor
extension fields, and unknown length list vs_data, which may contain
the version2 and/or MultiAP-Extension items.

son_slave_thread.cpp also needs to be modified. Now that the vendor
extension uses an unknown length list, it no longer contains the
default value for the Version2 attribute, so they have to be added
manually.

Fixes #490
Fixes #213

Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>
Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

arnout commented on framework/tlvf/yaml/tlvf/WSC/WSC_Attributes.yaml:
```diff
@@ -265,6 +265,18 @@ sWscAttrVersion2:
     _type: uint8_t
     _value: WSC_VERSION2
 
+sWscAttrVersion2:
```
It's not an attribute, it's a vendor extension IE. So `sWscIeVersion2` would be better. (nitpick)

With that, it is also no longer needed to change the name of sWscAttrVersion2.

arnout commented on framework/tlvf/yaml/tlvf/WSC/WSC_Attributes.yaml:
```diff
@@ -395,3 +407,25 @@ cWscAttrEncryptedSettings:
     _type: char
     _length: []
+
+cWscVendorExt:
```
Since the WFA ID is hardcoded here, it would be better to name it `cWscVendorExtWfa`.

arnout reviewed (approved):
Nitpick comment, and second commit needs an actual commit message, otherwise LGTM.

Commit:
tlvf: Add device name to WSC M2 TLV

Device name attribute was missing in our M2 TLV which caused the MDK
agent to reject our M2. According to table 9 of WSC spec the device name
attribute is required in M2, so add it.

This fixes autoconfiguration with MDK agent.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>
